src/client/liveblog.ts

```diff
@@ -1,45 +1,50 @@
 // ----- Imports ----- //
 
-import setup from 'client/setup';
-import { fromCapiLiveBlog, getFormat } from 'item';
 import ReactDOM from 'react-dom';
-import LiveblogBody from 'components/liveblog/body';
 import { createElement as h } from 'react';
-import { Content } from 'mapiThriftModels';
-import { parse } from './parser';
+import { Format, Design, Display, Pillar } from '@guardian/types/Format';
 
+import setup from 'client/setup';
+import { fromSerialisable } from 'liveBlock';
+import { parse } from 'client/parser';
+import LiveblogBody from 'components/liveblog/body';
 
-// ----- Run ----- //
 
-declare global {
-    interface Window {
-        __INITIAL__DATA__: Content;
-    }
+// ----- Setup ----- //
+
+const domParser = new DOMParser();
+
+const format: Format = {
+    design: Design.Live,
+    display: Display.Standard,
+    pillar: Pillar.News,
 }
 
-setup();
 
-const browserParser = (html: string): DocumentFragment =>
-    parse(new DOMParser())(html)
+// ----- Functions ----- //
+
+const docParser = (html: string): DocumentFragment =>
+    parse(domParser)(html)
         .toOption()
-        .withDefault(new DocumentFragment())
+        .withDefault(new DocumentFragment());
+
+const deserialise = fromSerialisable(docParser);
 
 
-try {
-    const hydrationProps = JSON.parse(document.getElementById('hydrationProps')?.textContent ?? "");
+// ----- Run ----- //
+
+// Set up interactives, Twitter embeds etc.
+setup();
 
-    if (hydrationProps) {
-        const item = fromCapiLiveBlog({
-            docParser: browserParser,
-            salt: 'mockSalt',
-        })(hydrationProps);
-        const { blocks, totalBodyBlocks } = item;
+// Load the initial group of blocks.
+fetch(`${window.location}/live-blocks`)
+    .then(res => res.json())
+    .then(({ newBlocks }) => {
+        const blocks = deserialise(newBlocks);
 
-        ReactDOM.hydrate(
-            h(LiveblogBody, { format: getFormat(item), blocks, totalBodyBlocks }),
-            document.querySelector('#blocks'),
+        ReactDOM.render(
```

Comment:
How comes we're no longer hydrating?

Reply:
At the moment because technically the data returned by `/live-blocks` might not be the same as what the server has rendered. There are a few sources of delay between the time the HTML is rendered on the server and the time the `fetch` request goes out:

- App caching
- Fastly caching
- Document request/response/render cycle

Liveblogs move quickly, and blocks may have been changed or new blocks added in that time. It might be that this is fine, or it might be that we delay this request until the first polling interval, or it might be best to store the initial few blocks using the `application/json` mechanism used currently. I haven't quite figured it out yet, but this implementation is pretty minimal at the moment and will probably change. There will _have_ to be a mechanism for accessing the `Format` and total block count regardless, so it might be that the mechanism used for that is used for this too.

Reply:
We will still SSR the initial blocks though? so a cached home front can still display those offline.

Reply:
The initial blocks are still rendered on the server, yes.